packages/cheqd/src/dids/didCheqdUtil.ts

```diff
@@ -0,0 +1,152 @@
+      case VerificationMethods.Ed255192018:
+        return vm.publicKeyBase58 != null
+      default:
+        return false
+    }
+  })
+
+  if (!isValidVerificationMethod) return { valid: false, error: 'verificationMethod publicKey is Invalid' }
+
+  const isValidService = didDocument.service
+    ? didDocument?.service?.every((s) => {
+        return s?.serviceEndpoint && s?.id && s?.type
+      })
+    : true
+
+  if (!isValidService) return { valid: false, error: 'Service is Invalid' }
+  return { valid: true } as SpecValidationResult
+}
+
+// Create helpers in sdk like MsgCreateDidDocPayload.fromDIDDocument to replace the below
+export async function createMsgCreateDidDocPayloadToSign(didPayload: DIDDocument, versionId: string) {
+  didPayload.service = didPayload.service?.map((e) => {
+    return {
+      ...e,
+      serviceEndpoint: Array.isArray(e.serviceEndpoint) ? e.serviceEndpoint : [e.serviceEndpoint],
+    }
+  })
+  const { protobufVerificationMethod, protobufService } = await DIDModule.validateSpecCompliantPayload(didPayload)
+  return MsgCreateDidDocPayload.encode(
+    MsgCreateDidDocPayload.fromPartial({
+      context: <string[]>didPayload?.['@context'],
+      id: didPayload.id,
+      controller: <string[]>didPayload.controller,
+      verificationMethod: protobufVerificationMethod,
+      authentication: <string[]>didPayload.authentication,
+      assertionMethod: <string[]>didPayload.assertionMethod,
+      capabilityInvocation: <string[]>didPayload.capabilityInvocation,
+      capabilityDelegation: <string[]>didPayload.capabilityDelegation,
+      keyAgreement: <string[]>didPayload.keyAgreement,
+      service: protobufService,
+      alsoKnownAs: <string[]>didPayload.alsoKnownAs,
+      versionId,
+    })
+  ).finish()
+}
+
+export function createMsgDeactivateDidDocPayloadToSign(didPayload: DIDDocument, versionId?: string) {
+  return MsgDeactivateDidDocPayload.encode(
+    MsgDeactivateDidDocPayload.fromPartial({
+      id: didPayload.id,
+      versionId,
+    })
+  ).finish()
+}
+
+export type SpecValidationResult = {
+  valid: boolean
+  error?: string
+}
+
+export function generateDidDoc(options: IDidDocOptions) {
+  const { verificationMethod, methodSpecificIdAlgo, verificationMethodId, network, publicKey } = options
+  const verificationKeys = createVerificationKeys(publicKey, methodSpecificIdAlgo, verificationMethodId, network)
+  if (!verificationKeys) {
+    throw new Error('Invalid DID options')
+  }
+  const verificationMethods = createDidVerificationMethod([verificationMethod], [verificationKeys])
+
+  return createDidPayload(verificationMethods, [verificationKeys]) as DidDocument
+}
+
+export interface IDidDocOptions {
+  verificationMethod: VerificationMethods
+  verificationMethodId: TVerificationKey<string, number>
+  methodSpecificIdAlgo: MethodSpecificIdAlgo
+  network: CheqdNetwork
+  publicKey: string
+}
+
+export function getClosestResourceVersion(resources: Metadata[], date: Date) {
```

Comment:
Let's add some comments here that this methods should be changed to particular queries when they will be ready on did-resolver side?